Enabling different slo endpoint for responses to IdP-initiated logout

BeritJanssen opened the pull request:
This PR addresses issue #13 , which was not yet correctly adressed by the library. Implemented the suggested fix. Additionally, the `OneLogin_Saml2_Logout_Response.build` method was modified (only called by the IdP-initiated logout case of `process_slo`) such that the `Destination` field of the response payload will be set from the `responseUrl` setting, if applicable.

Adjusted README and added two simple tests.

pitbulk commented:
HI @BeritJanssen,

Thanks for your contribution.

I plan to fix this as I fixed it at php-saml:
https://github.com/onelogin/php-saml/commit/7aca365b538b19d9dc3da8a36380e60d5eac5330

Is similar to your fix, but defines new methods in the settings class to later be used on logout_request and logout_response
Are you able to adapt your PR?

BeritJanssen commented:
Hi @pitbulk , I implemented the changes you suggested:
- I moved `get_slo_url`, `get_slo_return_url` and `get_sso_url` to the settings module
- I adjusted the tests (passing)
- I also adjusted `.html` files referencing the `get_sso_url` method so that they point to the settings rather than auth module

pitbulk commented:
I will review and merge asap.

Thanks for your contribution!

nltommynl commented:
@pitbulk, do you see an opportunity to merge this PR? This solves a lot of problems with clients using this library in combination with NetIQ Access Manager (and in our case there are quite a few 😊).

pitbulk commented:
Sorry for the delay.

Yes, I'm working right now merging PR and fixing pending issues on python-saml and python3-saml and I plan to have an official release later today

nltommynl commented:
@pitbulk perfect, thanks!

pitbulk commented:
I merged your changes, but applied some modifications:
https://github.com/onelogin/python3-saml/commit/9fdb11d1cd0190cfc783306cf9aed5133121a1ce

instead:
https://github.com/onelogin/python3-saml/pull/229/commits/9ca31059c5e79dbacba982d3cb21594ea4965488
